### ISSUE TITLE
Users should get information on what to do for Error Ill-formed WAVE file: no RIFF tag found 

### ISSUE THREAD

**Issue by pleira** [martinber/noaa-apt#20]
My [Satnogs Station 771](https://network.satnogs.org/stations/771/) is configured to store the IQ files of the different satellite passes. I have tried using noaa-apt on the saved files, but it does not work and an error message is produced instead:
`ERROR [noaa_apt] Ill-formed WAVE file: no RIFF tag found`

So this issue is really to have a way to use noaa-apt with these kind of IQ files. Maybe this could be done in noaa-apt code or maybe it could be documented how IQs should be prepared. In that case,  after and before the observations in satnogs, custom shell scripts can be run, which know details like TLE,  can be used to provide some information for noaa-apt. In any case, it would be great if users get some feedback on what they can do in this case.

The IQs should be accessible under the URL http://satnogs771.freedombox.rocks/, it might be not always available pending to fix Dynamic DNS updates.

**Comment by martinber** [comment]
Yes, this program can't demodulate IQ files. To do that, a FM demodulator should be implemented. Currently, only WAV files are supported because these are already FM demodulated by SDR#, GQRX, etc.

We could add documentation on how to demodulate IQ into WAV, but it is not very easy. The only way I know would be to open the IQ files in SDR# or GQRX, but there are different IQ formats, some work in some programs and not in others. As far as I know it would be hard to document because different steps are needed for different IQ files. I think that sox is the best tool to convert IQ files to different formats.

If you think that it is common to use Satnogs IQ files, we can document on how to convert with sox, open them in GQRX and export to WAV from there.

rtl_fm would be ideal for this, but I think it does not support loading of IQ files.

**Comment by pleira** [comment]
Thanks for the interesting feedback. I have renamed my IQ files to use raw as extension. 

Regarding documenting using rtl_fm and/or sox for preparing IQ files for noaa_apt, if simple to document, it might be worth to add a small description. Satnogs is going to be there producing IQ files in the future. This [weather-satellites](https://github.com/pfranchini/weather-satellites) project, has a [submit job](https://github.com/pfranchini/weather-satellites/blob/master/submit_job.sh) script which uses rtl_fm, and apt.sh which does `sox ${file}.wav -r 11025 ${file}_res.wav`. Satnogs itself, is using [gnuradio companion scripts](https://gitlab.com/librespacefoundation/satnogs/satnogs-flowgraphs/-/tree/master/satellites). I lack expertise for the details, but well, I welcome a recipe if it were a few shell commands.

**Comment by martinber** [comment]
I've added a short comment about this in the usage page.

This project looks really promising: https://github.com/cubehub/demod, as I can see, that program is able to demodulate any IQ file into WAV

The problem with IQ files is that there are different formats and it isn't evident from the filename. WAV files have this information inside the file (metadata) but with IQ files you have to know:

- If it has float or integer numbers
- The bit depth: 8 bit, 16 bit, etc.
- The sample rate
- Byte order: It is always little endian?
- Number of channels: 1 or 2 if stereo

To solve this issue, a short tutorial about this program would be needed. And also a list of common IQ formats, e.g. GQRX uses 32bit floats, little endian, 1 or 2 channels and the sample rate is listed in the filename
